File: python/src/calculator.py
"""
Minimal Calculator - アプリケーションコード（テスト対象）

このファイルがBDDでテストされる「実際のアプリケーション」です。
"""


class Calculator:
    def __init__(self):
        self.result = 0

    def add(self, value):
        self.result += value
        return self.result

    def subtract(self, value):
        self.result -= value
        return self.result

    def multiply(self, value):
        self.result *= value
        return self.result

    def clear(self):
        self.result = 0
        return self.result

File: javascript/src/calculator.js
/**
 * Minimal Calculator - アプリケーションコード（テスト対象）
 *
 * このファイルがBDDでテストされる「実際のアプリケーション」です。
 */

class Calculator {
  constructor() {
    this.result = 0;
  }

  add(value) {
    this.result += value;
    return this.result;
  }

  subtract(value) {
    this.result -= value;
    return this.result;
  }

  multiply(value) {
    this.result *= value;
    return this.result;
  }

  clear() {
    this.result = 0;
    return this.result;
  }
}

module.exports = { Calculator };
